packages/hardhat-chai-matchers/src/internal/addressable.ts

```diff
@@ -0,0 +1,71 @@
+import { isAddress, isAddressable } from "ethers";
+
+export function supportAddressable(
+  Assertion: Chai.AssertionStatic,
+  chaiUtils: Chai.ChaiUtils
+) {
+  const equalsFunction = override("eq", "equal", "not equal", chaiUtils);
+  Assertion.overwriteMethod("equals", equalsFunction);
+  Assertion.overwriteMethod("equal", equalsFunction);
+  Assertion.overwriteMethod("eq", equalsFunction);
+}
+
+type Methods = "eq";
+
+function override(
+  method: Methods,
+  name: string,
+  negativeName: string,
+  chaiUtils: Chai.ChaiUtils
+) {
+  return (_super: (...args: any[]) => any) =>
+    overwriteAddressableFunction(method, name, negativeName, _super, chaiUtils);
+}
+
+// ethers's Addressable have a .getAddress() that returns a Promise<string>. We don't want to deal with async here,
+// so we are looking for a sync way of getting the address. If an address was recovered, it is returned as a string,
+// otherwise undefined is returned.
+function tryGetAddress(value: any): string | undefined {
+  if (isAddress(value)) {
+    return value;
+  } else if (isAddressable(value)) {
+    return tryGetAddress((value as any).address ?? (value as any).target);
```

Comment:
Just realized [`.target` can also be an ENS name](https://docs.ethers.org/v6/api/contract/#BaseContract-target), we should test also with those cases

Reply:
If target is an ENS name, doing `tryGetAddress(value.target)` will return `undefined` (because `isAddress()` returns false on ENS).

AFAIK, this is the best we can do without an async resolution.

Reply:
Just double checked and you're right. Agree this seems to be the best we can do